src/java/org/apache/cassandra/schema/AutoRepairParams.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.schema;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang3.StringUtils;
+
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.repair.autorepair.AutoRepairConfig;
+import org.apache.cassandra.utils.LocalizeString;
+
+import static java.lang.String.format;
+import static org.apache.cassandra.utils.LocalizeString.toLowerCaseLocalized;
+
+/**
+ * AutoRepair table parameters - used to define the auto-repair configuration for a table.
+ */
+public final class AutoRepairParams
+{
+    public enum Option
+    {
+        FULL_ENABLED,
+        INCREMENTAL_ENABLED,
+        PREVIEW_REPAIRED_ENABLED,
+        PRIORITY;
+
+        @Override
+        public String toString()
+        {
+            return toLowerCaseLocalized(name());
+        }
+    }
+
+    private final ImmutableMap<String, String> options;
+
+    public static final Map<String, String> DEFAULT_OPTIONS = ImmutableMap.of(
+    LocalizeString.toLowerCaseLocalized(Option.FULL_ENABLED.name()), Boolean.toString(true),
+    LocalizeString.toLowerCaseLocalized(Option.INCREMENTAL_ENABLED.name()), Boolean.toString(true),
+    LocalizeString.toLowerCaseLocalized(Option.PREVIEW_REPAIRED_ENABLED.name()), Boolean.toString(true),
+    Option.PRIORITY.toString(), "0"
+    );
+
+    AutoRepairParams(Map<String, String> options)
+    {
+        this.options = ImmutableMap.copyOf(options);
+    }
+
+    public static final AutoRepairParams DEFAULT =
+    new AutoRepairParams(DEFAULT_OPTIONS);
+
+    public static AutoRepairParams create(Map<String, String> options)
+    {
+        Map<String, String> optionsMap = new TreeMap<>();
+        for (Map.Entry<String, String> entry : DEFAULT_OPTIONS.entrySet())
+        {
+            optionsMap.put(entry.getKey(), entry.getValue());
+        }
```

Comment:
NIT, we can initialize this in the constructor directly:
```suggestion
        Map<String, String> optionsMap = new TreeMap<>(DEFAULT_OPTIONS);
```